src/distributions/weighted_index.rs

```diff
@@ -256,6 +255,30 @@ mod test {
         assert_eq!(de_weighted_index.total_weight, weighted_index.total_weight);
     }
 
+    #[test]
+    fn test_accepting_nan(){
+        assert_eq!(
+            WeightedIndex::new(&[std::f32::NAN, 0.5]).unwrap_err(),
```

Comment:
I think this fails our `no_std` build. Maybe try `core::f32::NAN`?

Reply:
Makes sense!